package-res/reportviewer/reportviewer-prompt.js

```diff
@@ -93,7 +93,7 @@ define([
               // add to changed
               changedParameters.push(parameter.name);
             }
-          } else {
+          } else if("" != parameter.value) {
```

Comment:
parameter.value can also legally be <null> in some cases. The server would produce an XML element "value" with null="true" as attribute. (see line 771 in ParameterXmlContentHandler). So unless the UI normalizes all occurences of <null> to "", I would extend the test here.